frontend/src/reducers/failures/details/index.js

```diff
@@ -0,0 +1,32 @@
+import {
+  FAILURE_SHOW_RETRY,
+  FAILURE_HIDE_RETRY,
+  FAILURE_HIDE_OVERVIEW,
+  RECEIVE_FAILURE_DETAILS
+} from 'actions'
+
+function patchFailure (state, action, params) {
+  if (state.id === action.failure.id) {
+    return Object.assign(state, params)
```

Comment:
I think it's better to use the expand object syntax, so that we don't mutate the params given:
```javascript
{ ...state, ...params }
```